### PR TITLE
[Draft] Orbit SDK docs (TW edits)

### PR DESCRIPTION
**Pull request by anegg0**
This version contains various edits:
- Re-linkings (to avoid `here` and `there` links)
- Rewording
- `quicklook` reference links
- Reformats

### CODE REVIEW

**Comment by symbolpunk** [comment]
In review ^^

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+title: 'How to configure an Orbit chain using the Orbit SDK'
+sidebar_label: 'Orbit Chain Configuration'
+description: 'How to configure an Orbit chain using the Orbit SDK '
+author: Mehdi Salehi
+sme: Mehdi Salehi
+target_audience: 'Developers deploying and maintaining Orbit chains.'
+sidebar_position: 6
+---
+
+This guide will explore the essentials of configuring your Orbit chain. Configuration includes a range of settings from the parent chain and node configuration to specific child chain parameter configurations. Our focus here is on child chain configuration, guiding you through the necessary adjustments to customize your chain's operation.
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:11]:
```suggestion
In this how-to, you'll learn how to configure your Orbit chain using the Orbit SDK. This includes **parent chain**, **node**, and **child chain parameter** configuration.
```

- nit - "introduce the document as its content type explicitly" (undocumented pattern, added to pattern backlog [here](https://www.notion.so/arbitrum/Pattern-guide-Docs-dafaf8d1ce34433d88c1ef3b4b99c19c?pvs=4#a815de55df444b98ba8ac990424a1353))
- nit - consistent section labeling between this intro, and the header titles down below

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+In the introductory section, we outlined three primary configurations required for setting up and managing an Orbit chain. These configurations play a crucial role in ensuring the efficient operation of your chain. Let's delve into each type:
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:13]:
```suggestion
```

weakly held opinion - we can cut this and dive into the three configs

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+## 1. Parent Chain Configuration
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:15]:
```suggestion
## 1. Parent chain configuration
```

nit - sentence-case titles of all kinds ([ref](https://docs.arbitrum.io/for-devs/contribute#style-conventions))

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+Configuring the parent chain is an essential initial step in setting up your Orbit chain. Most of these configurations are specified during the setup phase. Detailed instructions can be found in the [Rollup Deployment Parameters](deploying-rollup-chain-with-sdk#rollup-deployment-parameter) section of the rollup deployment guide. 
+
+After the initial setup, the chain owner can modify configurations as needed. For instance, the validator set can be updated by invoking the [`setValidKeyset`](https://github.com/OffchainLabs/nitro-contracts/blob/90037b996509312ef1addb3f9352457b8a99d6a6/src/bridge/SequencerInbox.sol#L751) function with a new set of validators. This adaptability facilitates continuous optimization and management of the chain.
+
+## 2. Node/Chain Configuration
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:20]:
```suggestion
## 2. Node configuration
```
- nit - sentence-case titles of all kinds ([ref](https://docs.arbitrum.io/for-devs/contribute#style-conventions))
- nit - consistent section labeling between header titles, and the intro up top

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+This category includes settings adjustable within the `nodeConfig.json` file, directly impacting the operation of the chain's nodes, including special nodes like validators and sequencers. These settings are vital for tailoring the node's functionality to specific requirements or performance criteria. The chain owner can modify these configurations during the node config generation process, ensuring that each node operates with the desired settings from the start. For more information, refer to the [Node Configuration Preparation](preparing-node-config-with-sdk.md) documentation.
+
+## 3. Child Chain Parameter Configuration
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:23]:
```suggestion
## 3. Child chain parameter configuration
```

nit - sentence-case titles of all kinds ([ref](https://docs.arbitrum.io/for-devs/contribute#style-conventions))

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+The final configuration type involves setting parameters on the child chain. This level of configuration is primarily achieved through the [ArbOwner precompile](https://github.com/OffchainLabs/nitro-contracts/blob/main/src/precompiles/ArbOwner.sol) on the child chain. These configurations are typically applied post-chain initialization and after the deployment of the token bridge. This guide will help you configure child chain parameters using the Orbit SDK, providing insights into effective management and optimization strategies for your child chain operations.
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:24]:
```suggestion
The final configuration type involves setting parameters on the child chain. This level of configuration is primarily achieved through the [ArbOwner precompile](https://github.com/OffchainLabs/nitro-contracts/blob/main/src/precompiles/ArbOwner.sol) on the child chain. These configurations are typically applied post-chain initialization and after the deployment of the token bridge.
```

weakly held opinion - this copy doesn't add value

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+The child chain configuration can be performed after the chain initialization. These parameters are configurable via setter functions in the [ArbOwner precompile](https://github.com/OffchainLabs/nitro-contracts/blob/main/src/precompiles/ArbOwner.sol). Additionally, there are various getter functions in the ArbOwner precompile that you can use to read the current configuration. Below, we explain several methods in the ArbOwner precompile that you can use to configure the parameters or read their current state.
+
+### Setter Functions
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:28]:
```suggestion
### Setter functions
```

nit - sentence-case titles of all kinds ([ref](https://docs.arbitrum.io/for-devs/contribute#style-conventions))

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+You can use these setter functions to configure the child chain parameters:
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:29]: This reads similar to reference content; to maintain consistency, we can reformat as a table. As a future optimization / separate PR, we can relocate this into a dedicated reference document (primarily "tables of things"), and then either link to it here, or partialize it and include it in both.

```markdown
| Method                | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
|-----------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `addChainOwner`       | Allows you to add a new chain owner to your Orbit chain.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               |
| `removeChainOwner`    | Enables you to remove an existing owner from the list of chain owners.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 |
| `setMinimumL2BaseFee` | Sets the minimum base fee on the child chain. The minimum base fee is the lowest amount that the base fee on the child chain can ever be. For example, the current minimum base fee on Arbitrum One is 0.1 gwei, and on Arbitrum Nova, it is 0.01 gwei.                                                                                                                                                                                                                                                                                                                                |
| `setSpeedLimit`       | The fee mechanism on the Arbitrum Nitro stack differs from the Ethereum blockchain. The Nitro stack has a parameter called the speed limit, which targets the number of gas consumed on the child chain per second. If the amount of gas per second exceeds this pre-specified amount, the base fee on the child chain will increase, and vice versa. The current speed limit on Arbitrum One is 7 million gas per second, meaning if the Arbitrum One chain consumes more than 7 million gas per second, its base fee will increase. For more information on the speed limit, please refer to this [document](https://docs.arbitrum.io/inside-arbitrum-nitro/#the-speed-limit). |
| `setInfraFeeAccount`  | Sets the infrastructure fee account address, which receives all fees collected on the child chain. It is meant to receive the minimum base fee, with any amount above that going to the network fee account.                                                                                                                                                                                                                                                                                                                                                                           |
| `setNetworkFeeAccount`| Sets the network fee account address. As mentioned, this address collects all fees above the base fee. Note that if you set this amount to the 0 address on your chain, all fees will be deposited into the infrastructure fee account.                                                                                                                                                                                                                                                                                                                                                 |
| `scheduleArbOSUpgrade`| If you plan to upgrade the <a data-quicklook-from="arbos">ArbOS</a> version of your chain, this method can help you schedule the upgrade. For a complete guide on this matter, please refer to the explanation of the [arbos upgrade](../../how-tos/arbos-upgrade.md).                                                                                                                                                                                                                                                                                                                   |
| `setChainConfig`      | We discussed the chainConfig in the [Rollup deployment guide](deploying-rollup-chain-with-sdk.md#chain-config-parameter) in detail. If you wish to change any field of the `chainConfig`, you need to use this method on the child chain.                                                                                                                                                                                                                                                                                                                                               |
````

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+
+- `setInfraFeeAccount`: Sets the infrastructure fee account address, which receives all fees collected on the child chain. It is meant to receive the minimum base fee, with any amount above that going to the network fee account.
+
+- `setNetworkFeeAccount`: Sets the network fee account address. As mentioned, this address collects all fees above the base fee. Note that if you set this amount to the 0 address on your chain, all fees will be deposited into the infrastructure fee account.
+
+- `scheduleArbOSUpgrade`: If you plan to upgrade the <a data-quicklook-from="arbos">ArbOS</a> version of your chain, this method can help you schedule the upgrade. For a complete guide on this matter, please refer to the explanation of the [arbos upgrade](../../how-tos/arbos-upgrade.md).
+
+- `setChainConfig`: We discussed the chainConfig in the [Rollup deployment guide](deploying-rollup-chain-with-sdk.md#chain-config-parameter) in detail. If you wish to change any field of the `chainConfig`, you need to use this method on the child chain.
+
+### Getter Functions
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:47]:
```suggestion
### Getter functions
```

nit - sentence-case titles of all kinds ([ref](https://docs.arbitrum.io/for-devs/contribute#style-conventions))

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+
+To read the child chain parameters, you can use these getter functions:
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:49]: This reads similar to reference content; to maintain consistency, we can reformat as a table. As a future optimization / separate PR, we can relocate this into a dedicated reference document (primarily "tables of things"), and then either link to it here, or partialize it and include it in both.

```markdown
| Method                 | Description                                               |
|------------------------|-----------------------------------------------------------|
| `getAllChainOwners`    | Provides the list of all current chain owners.            |
| `getNetworkFeeAccount` | Returns the network fee account address.                  |
| `getInfraFeeAccount`   | Returns the infrastructure fee account address.           |
| `isChainOwner`         | Allows you to check whether an address is on the list of chain owners. |
```

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+
+- `getAllChainOwners`: Provides the list of all current chain owners.
+
+- `getNetworkFeeAccount`:  Returns the network fee account address.
+
+- `getInfraFeeAccount`:  Returns the infrastructure fee account address.
+
+- `isChainOwner`:  Allows you to check whether an address is on the list of chain owners.
+
+### Configuring the Child Chain Using the Orbit SDK
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:59]:
```suggestion
### Configure the child chain using the Orbit SDK
```
- nit - sentence-case titles of all kinds ([ref](https://docs.arbitrum.io/for-devs/contribute#style-conventions))
- nit - imperative titles in how-tos (undocumented pattern - [ref](https://www.notion.so/arbitrum/How-to-775951f3d84c476cb6cba296a0a6f671?pvs=4))

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+const client = createPublicClient({
+  chain: arbitrumLocal,
+  transport: http(),
+}).extend(arbOwnerPublicActions);
+```
+
+With `arbOwnerPublicActions` and the public client in the Orbit SDK, we've added two new methods to the public clients:
+
+
+#### 1. arbOwnerReadContract
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:77]:
```suggestion
#### 1. `arbOwnerReadContract`
```

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+
+```js
+  const result = await client.arbOwnerReadContract({
+    functionName: 'getAllChainOwners',
+  });
+```
+
+Changing the function names in the list in this [section](#getter-functions) will give you the other parameters.
+
+#### 2. arbOwnerPrepareTransactionRequest
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:89]:
```suggestion
#### 2. `arbOwnerPrepareTransactionRequest`
```

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+
+This method can be used to configure the parameters on the ArbOwner precompile, which are listed in this [section](#setter-functions). An example of utilizing this method to configure parameters using the `client` defined in the previous section is:
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:91]:
```suggestion
This method can be used to configure the parameters on the ArbOwner precompile, which are listed in [the Setter functions section](#setter-functions). An example of utilizing this method to configure parameters using the `client` defined in the previous section is:
```

nit - specific link labels ([ref](https://docs.arbitrum.io/for-devs/contribute#style-conventions))

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 15]
```diff
@@ -0,0 +1,129 @@
...
+
+This method can be used to read the parameters of the child chain discussed in the [previous section](#getter-functions). An example of using this method with the `client` defined in the previous section is:
+
+```js
+  const result = await client.arbOwnerReadContract({
+    functionName: 'getAllChainOwners',
+  });
+```
+
+Changing the function names in the list in this [section](#getter-functions) will give you the other parameters.
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:87]:
```suggestion
Changing the function names in the list in [the Getter functions section](#getter-functions) will give you the other parameters.
```
nit - specific link labels ([ref](https://docs.arbitrum.io/for-devs/contribute#style-conventions))

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 16]
```diff
@@ -0,0 +1,129 @@
...
+    account: owner.address,
+  });
+
+  // Submitting the transaction to add a chain owner
+  await client.sendRawTransaction({
+    serializedTransaction: await owner.signTransaction(transactionRequest),
+  });
+```
+
+To use this method, as shown in the example above, some inputs need to be defined. `functionName` is the name of the method you want to use to set the parameter, which can be found in [this section](#setter-functions). `args` are the arguments for the defined function. The `upgradeExecutor` field specifies whether an `upgradeExecutor` contract governs your chain. If it is not using an `upgradeExecutor`, you can set it to `false`, similar to the example above. Also, the `account` field defines the chain owner if an `upgradeExecutor` does not govern the chain.
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:108]:
```suggestion
To use this method as shown in the example above, some inputs need to be defined:

| Parameter        | Description                                                                                                                                                                 |
|------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `functionName`   | The name of the method you want to use to set the parameter, which can be found in [the Setter functions section](#setter-functions).                                        |
| `args`           | The arguments for the defined function.                                                                                                                                     |
| `upgradeExecutor`| Specifies whether an `upgradeExecutor` contract governs your chain. If it is not using an `upgradeExecutor`, you can set it to `false`, similar to the example above.       |
| `account`        | Defines the chain owner if an `upgradeExecutor` does not govern the chain.                                                                                                  |
```

- nit - specific link labels ([ref](https://docs.arbitrum.io/for-devs/contribute#style-conventions))
- tables == happiness

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md [hunk 17]
```diff
@@ -0,0 +1,129 @@
...
+  });
+
+  // Submitting the transaction to add a chain owner
+  await client.sendRawTransaction({
+    serializedTransaction: await owner.signTransaction(transactionRequest),
+  });
+```
+
+
+In this example, all the fields are the same as in the first example, except the `upgradeExecutor` field, which you need to set to the `upgradeExecutor` address, and the `account` parameter, which needs to be set to the owner of the upgradeExecutor contract.
```
Comment (symbolpunk) [inline arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:128]:
```suggestion
In this example, all the fields are the same as in the first example, except the `upgradeExecutor` field, which you need to set to the `upgradeExecutor` address, and the `account` parameter, which needs to be set to the owner of the `upgradeExecutor` contract.
```

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md (hunk 3 above)
Comment (anegg0) [reply arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:14]: Missed this one entirely, thanks.

File: arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md (hunk 14 above)
Comment (anegg0) [reply arbitrum-docs/launch-orbit-chain/orbit-sdk/how-tos/configuring-orbit-chain-with-sdk.md:90]: It's quite shameful to have missed those. They were in my face!

**Comment by symbolpunk** [comment]
Marking as `Draft` while TW stabilizes

**Review by symbolpunk — APPROVED** [review]
LGTM; we can address any remaining TODOs in #845 directly

**Pull request merged**